### PR TITLE
Embedded search

### PR DESCRIPTION
**Pull request by harunurhan**

### CODE REVIEW

File: src/common/components/EmbeddedSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    this.updateQueryStateAndSearch({ [aggregationKey]: selections });
+  }
+
+  // eslint-disable-next-line react/sort-comp
+  async updateQueryStateAndSearch(partialQuery) {
+    const query = {
+      ...this.state.query,
+      ...partialQuery,
+    };
+    await this.setState({ query });
```
Comment (drjova) [inline src/common/components/EmbeddedSearch.jsx]: why we need here the await?

File: src/common/components/EmbeddedSearch.jsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+    this.state = {
+      query: {
+        page: 1,
+        size: 10,
+        sort: 'mostrecent',
+      },
+      numberOfResults: 0,
+      aggregations: Map(),
+      results: List(),
```
Comment (drjova) [inline src/common/components/EmbeddedSearch.jsx]: do we really need immutable here?

File: src/common/components/EmbeddedSearch.jsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    const {
+      query,
+      aggregations,
+      results,
+      numberOfResults,
+      loadingAggregations,
+      loadingResults,
+    } = this.state;
+    return (
+      this.renderErrorOrNull() || (
```
Comment (drjova) [inline src/common/components/EmbeddedSearch.jsx]: shall we create a component to reuse it? We'll need for other places as well

File: package.json [hunk 4]
```diff
@@ -30,8 +30,10 @@
     "immutable": "^3.8.2",
     "is-valid-orcid": "^1.2.0",
     "jsondiffpatch": "^0.3.11",
+    "lodash.clonedeep": "^4.5.0",
     "lodash.isempty": "^4.4.0",
     "lodash.isplainobject": "^4.0.6",
+    "lodash.mergewith": "^4.6.1",
```
Comment (drjova) [inline package.json]: do we still need this?

File: src/common/components/EmbeddedSearch.jsx (hunk 3 above)
Comment (harunurhan) [reply src/common/components/EmbeddedSearch.jsx]: I don't think it's necessary, at the end it just an `Alert` and message is already specific for this use anyway.

Comment (drjova) [reply src/common/components/EmbeddedSearch.jsx]: it's nice to use in places such as references, citations etc

**Pull request merged**